### PR TITLE
refactor: `& Impure` to `\ IO`

### PR DESCRIPTION
**Pull request by ninaandrup**
Related to https://github.com/flix/flix/issues/4377

### CODE REVIEW

**Comment by ninaandrup** [comment]
Not all the examples seem to be fully updated.

There is still a function with `Impure` effect in this example: [select-with-defaults-and-timers](https://github.com/flix/flix/blob/b85ac87275b01e6cae567097e0300c38e3ab81fe/examples/select-with-defaults-and-timers.flix#L27)

And both example [effect-polymorphic-functions](https://github.com/flix/flix/blob/master/examples/effect-polymorphic-functions.flix) and [pure-and-impure-functions](https://github.com/flix/flix/blob/master/examples/pure-and-impure-functions.flix) have descriptions using the `Impure` keyword.

These things should still be updated on the website, right?

**Comment by ninaandrup** [comment]
How should the [description of pure and impure](https://github.com/ninaandrup/flix.dev/blob/f601ad5c07725ec78c1d86c34e77ce9e1ec291cd/src/page/About.js#L179) code be rewritten? Do we call it `IO effect` now? Or is it still called `Impure` code even though we use a new keyword in the code?

**Comment by magnus-madsen** [comment]
> Not all the examples seem to be fully updated.
> 
> There is still a function with `Impure` effect in this example: [select-with-defaults-and-timers](https://github.com/flix/flix/blob/b85ac87275b01e6cae567097e0300c38e3ab81fe/examples/select-with-defaults-and-timers.flix#L27)
> 
> And both example [effect-polymorphic-functions](https://github.com/flix/flix/blob/master/examples/effect-polymorphic-functions.flix) and [pure-and-impure-functions](https://github.com/flix/flix/blob/master/examples/pure-and-impure-functions.flix) have descriptions using the `Impure` keyword.
> 
> These things should still be updated on the website, right?

I think that should be fixed by Matt soon.

**Comment by magnus-madsen** [comment]
> How should the [description of pure and impure](https://github.com/ninaandrup/flix.dev/blob/f601ad5c07725ec78c1d86c34e77ce9e1ec291cd/src/page/About.js#L179) code be rewritten? Do we call it `IO effect` now? Or is it still called `Impure` code even though we use a new keyword in the code?

We should use phrases like: "effectful function". "function with IO effect" etc. We can also still say "impure" in text, if it makes sense.

File: src/page/About.js [hunk 1]
```diff
@@ -176,11 +176,11 @@ def main(): Unit & Impure =
     } |> println`}
                         </InlineEditor>
 
-                        <h2>Polymorphic Effects: Separating Pure and Impure Code</h2>
+                        <h2>Polymorphic Effects: Separating Pure and Effectful Code</h2>
 
                         <p>
                             A unique feature of Flix is its polymorphic effect system. The Flix type and effect system
-                            cleanly separates pure and impure code. If an expression is pure then it always
+                            cleanly separates pure and effectful code. If an expression is pure then it always
```
Comment (ninaandrup) [inline src/page/About.js:183]: I do not know, if it makes sense to just still write `impure`?

Comment (ninaandrup) [reply src/page/About.js:183]: I'm not even sure if it is correct to use effectful here? Is an `effectful function` the same as an `impure function` and a `functions with IO effect`?

Comment (magnus-madsen) [reply src/page/About.js:183]: I think here its fine to say separate pure and impure code.

Comment (magnus-madsen) [reply src/page/About.js:183]: an impure function and an effectful function is more or less the same.

File: src/page/blog/PolymorphicEffects.js [hunk 2]
```diff
@@ -453,7 +453,7 @@ List.map(f, List.map(g, 1 :: 2 :: Nil))`}
                         </p>
 
                         <InlineEditor>
-                            {`def mapCompose(f: a -> b & e1, g: b -> c & {{(not e1) \\/ e2}}, xs: List[a]): ... = ...`}
+                            {`def mapCompose(f: a -> b \\ e1, g: b -> c \\ {{(not e1) or e2}}, xs: List[a]): ... = ...`}
```
Comment (ninaandrup) [inline src/page/blog/PolymorphicEffects.js:456]: I think I will need some help to make this correct:
```
def mapCompose(f: a -> b \ e1, g: b -> c \ {{(not e1) or e2}}, xs: List[a]): ... = ...
```

File: src/page/blog/Naming.js [hunk 3]
```diff
@@ -91,7 +91,7 @@ class Naming extends Component {
                         </p>
 
                         <InlineEditor>
-                            {`def mapInPlace(f: a -> a, a: Array[a]): Unit & Impure`}
+                            {`def mapInPlace(f: a -> a \\ ef, a: Array[a]): Unit \\ IO`}
```
Comment (ninaandrup) [inline src/page/blog/Naming.js:94]: Would `mapInPlace` be impure, if `f` is pure?

Comment (ninaandrup) [reply src/page/blog/Naming.js:94]: Given that `mapInPlace` returns `Unit`

Comment (magnus-madsen) [reply src/page/blog/Naming.js:94]: Yes, it should be IO as you have it.

**Pull request merged**